pygmt/helpers/decorators.py

```diff
@@ -338,9 +357,15 @@ def new_module(*args, **kwargs):
                     value = kwargs[arg]
                     issequence = fmt in separators
                     if issequence and is_nonstr_iter(value):
-                        kwargs[arg] = separators[fmt].join(
-                            "{}".format(item) for item in value
-                        )
+                        for index, item in enumerate(value):
```

Comment:
Not sure if I understand your codes correctly. It seems the codes check all arguments. For example, **perspective=[30, 50]** is converted to **-p30/50**, and the codes also check if `30` and `50` are datetime-like, right?

However, **-p** doesn't accept any datetime-like values. It looks a waste of time. Perhaps we should check whether `arg="B"` (maybe arg="region")?

Reply:
> Not sure if I understand your codes correctly. It seems the codes check all arguments. For example, **perspective=[30, 50]** is converted to **-p30/50**, and the codes also check if `30` and `50` are datetime-like, right?

Yes, we do check every item using `assert " " not in str(item)`, but this is quite a fast string parsing check (see also the other comment on EAFP syntax). It is *not* explictily checking for datetime-like (which *would* be a slower check) using something like `isinstance(item, (pd.Timestamp, datetime.datetime, etc))`.

> However, **-p** doesn't accept any datetime-like values. It looks a waste of time. Perhaps we should check whether `arg="B"` (maybe arg="region")?

This would be an extra check on top of the (rather fast) `assert " " not in str(item)`, and yes, we could add it I suppose. But really, there's usually not very many items to parse here, only 4 if using `[xmin, xmax, ymin, ymax]`, maybe 6 if using `[xmin, xmax, ymin, ymax, zmin, zmax]`.

Happy to consider an alternative way though if you have any ideas, I definitely think there's a better way to do this. The unit tests are all there, so feel free to refactor/change the code as long as it passes the tests!